app/controllers/course/assessment/submissions_controller.rb

```diff
@@ -24,6 +24,14 @@ def edit
   end
 
   def update
+    if params[:submit_answer]
```

Comment:
Think that won't fix it ? and I am actually grading the answers when they are submitted.

This action was for submitting an answer, and the `else` branch was for only saving the answer(or submission).  That's the complex part and also the reason I need a `if else`